consensus/src/experimental/buffer_manager.rs

```diff
@@ -731,6 +734,18 @@ impl BufferManager {
                             self.advance_signing_root().await;
                         }
                     });
+                }
+                commit_msg = self.commit_msg_rx.select_next_some() => {
+                    let epoch_state = self.epoch_state.clone();
+                    let tx = verified_commit_msg_tx.clone();
+                    bounded_executor.spawn_blocking(move || {
+                        match commit_msg.req.verify(&epoch_state.verifier) {
+                            Ok(_) => {
+                                let _ = tx.unbounded_send(commit_msg);
+                            },
+                            Err(e) => warn!("Invalid commit message: {}", e),
+                        }
+                    }).await;
```

Comment:
this awaits in place so won't free the main loop?

Reply:
Okay, got it, `BoundedExecutor::spawn_block()` has different API than `tokio::spawn_blocking()`

Reply:
which is mildly unfortunate :trollface:

Reply:
isn't it actually blocking, because semaphore is being awaited on? if the semaphore is all taken, then it will block this loop, so we will never deque from the verified_rx?